plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2925,7 +2925,7 @@ public ConnectorTableHandle makeCompatiblePartitioning(ConnectorSession session,
     static TupleDomain<ColumnHandle> createPredicate(List<ColumnHandle> partitionColumns, List<HivePartition> partitions)
     {
         if (partitions.isEmpty()) {
-            return TupleDomain.none();
+            return partitionColumns.isEmpty() ? TupleDomain.none() : TupleDomain.all();
```

Comment:
+ originally we return TupleDomain.none() when partitions.isEmpty()
+ but in the case of no data of the PartitionedTable, we will throw IllegalArgumentException in `IoPlanPrinter.parseConstraints`. because the constraint is none.
+ for no data of the PartitionedTable case, the partitions is empty, and partitionColumns is not empty

Reply:
thanks for the explanation

> * but in the case of no data of the PartitionedTable, we will throw IllegalArgumentException in `IoPlanPrinter.parseConstraints`. because the constraint is none.

does it mean we should fix that method instead?